core/network/address.go

```diff
@@ -307,6 +319,28 @@ func NewMachineAddress(value string, options ...func(AddressMutator)) MachineAdd
 	return addr
 }
 
+// MachinesAddresses is a slice of MachineAddress
+type MachineAddresses []MachineAddress
+
+func (as MachineAddresses) AsProviderAddresses(options ...func(mutator ProviderAddressMutator)) ProviderAddresses {
+	addrs := make(ProviderAddresses, len(as))
+	for i, addr := range as {
+		addrs[i] = addr.AsProviderAddress(options...)
+	}
+	return addrs
+}
+
+// NewMachineAddresses is a convenience function to create addresses
+// from a variable number of string arguments, applying any supplied
+// options to each address
+func NewMachineAddresses(values []string, options ...func(AddressMutator)) MachineAddresses {
```

Comment:
Admittedly using a slice of strings was a compromise. I decided that it would be better to keep `options` as `...func()` consistent with the other constructors, meaning `values` needed to be changed to a slice.

If you can think of a better way I'd be happy to hear it :slightly_smiling_face: